core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -435,20 +436,167 @@ ClientRequestContext newDerivedContext(RequestId id, @Nullable HttpRequest req,
      */
     long responseTimeoutMillis();
 
+    /**
+     * Clears the response timeout.
+     */
+    void clearResponseTimeout();
+
     /**
      * Sets the amount of time allowed until receiving the {@link Response} completely
      * since the transfer of the {@link Response} started. This value is initially set from
      * {@link ClientOption#RESPONSE_TIMEOUT_MILLIS}.
+     *
+     * <p>For example:
+     * <pre>{@code
+     * ClientRequestContext ctx = ...;
+     * ctx.setResponseTimeoutMillis(1000);
+     * assert ctx.responseTimeoutMillis() == 1000;
+     * // Do something...
+     * ctx.setResponseTimeoutMillis(2000);
+     * assert ctx.responseTimeoutMillis() == 2000;
+     * }</pre>
+     *
+     * @param responseTimeoutMillis the amount of time allowed in milliseconds from the start time
+     *                              of the request
+     *
+     * @deprecated Use {@link #extendResponseTimeoutMillis(long)}, {@link #setResponseTimeoutAfterMillis(long)}
+     *             or {@link #setResponseTimeoutAfterMillis(long)}
      */
+    @Deprecated
     void setResponseTimeoutMillis(long responseTimeoutMillis);
 
     /**
      * Sets the amount of time allowed until receiving the {@link Response} completely
      * since the transfer of the {@link Response} started. This value is initially set from
      * {@link ClientOption#RESPONSE_TIMEOUT_MILLIS}.
+     *
+     * <p>For example:
+     * <pre>{@code
+     * ClientRequestContext ctx = ...;
+     * ctx.setResponseTimeout(Duration.ofSeconds(1));
+     * assert ctx.responseTimeoutMillis() == 1000;
+     * // Do something...
+     * ctx.setResponseTimeout(Duration.ofSeconds(2));
+     * assert ctx.responseTimeoutMillis() == 2000;
+     * }</pre>
+     *
+     * @param responseTimeout the amount of time allowed from the start time of the request
+     *
+     * @deprecated Use {@link #extendResponseTimeout(Duration)}, {@link #setResponseTimeoutAfter(Duration)}
+     *             or {@link #setResponseTimeoutAt(Instant)}
      */
+    @Deprecated
     void setResponseTimeout(Duration responseTimeout);
 
+    /**
+     * Extends the existing response timeout allowed by the specified {@code adjustmentMillis} until
+     * receiving the {@link Response} completely since the transfer of the {@link Response} started.
```

Comment:
How about this:

```
Extends the previously scheduled the response timeout by the specified amount of
{@code adjustmentMillis}. A new response timeout is scheduled if no response timeout
was scheduled previously and {@code adjustmentMillis} is a positive value.
```

Reply:
> A new response timeout is scheduled if no response timeout
was scheduled previously

I'm sure this is the right behavior. `no response timeout
was scheduled` means it has an infinite timeout. I think Extending or Reducing infinity is infinity. `infinity = extend(infinity + adjustment)`.

How about `The specified {@code adjustmentMillis} is ignored if the {@link Response} has no timeout`?

Reply:
Then, how about:

```
This method does nothing if no response timeout was scheduled previously.`
```

..?